src/core/commandRunner.ts

```diff
@@ -1,41 +1,53 @@
-import { execSync, ExecSyncOptions } from "child_process"
+import { exec, ExecOptions } from "child_process"
 import * as _ from "underscore"
-
 export interface ICommandRunner {
-  runCommand(command: string, options?: ExecSyncOptions): Promise<string>
+  runCommand(command: string, options?: ExecOptions): Promise<string>
+}
+
+export interface CustomLogger {
+  error(data: string): void
+  info(data: string): void
 }
 
 export class CommandRunner implements ICommandRunner {
-  constructor(private SFDXPath: string) {}
+  constructor(
+    private SFDXPath: string,
+    private useLiveLog: boolean = false,
+    private customLogger: CustomLogger = console
+  ) {}
 
-  public runCommand(
-    command: string,
-    options?: ExecSyncOptions
-  ): Promise<string> {
+  public runCommand(command: string, options?: ExecOptions): Promise<string> {
     let executePromise = new Promise<string>((resolve, reject) => {
       const fullCommand = this.SFDXPath + " " + command
 
-      let actualOptions: ExecSyncOptions = {
-        stdio: "pipe",
+      let actualOptions: ExecOptions = {
         env: this.getCommandEnv()
       }
 
       if (options !== undefined) {
         actualOptions = Object.assign(actualOptions, options)
       }
 
-      try {
-        let buffer = execSync(fullCommand, actualOptions)
-        resolve((buffer as Buffer).toString("utf8"))
-      } catch (e) {
-        let message: string
-        if (e !== undefined && e.stderr !== undefined) {
-          message = e["stderr"].toString("utf8")
+      let execProcess = exec(fullCommand, actualOptions, (error, stdout, stderr) => {
+        const errorMessage = stderr.toString()
```

Comment:
`;`

Reply:
👍 I'll add enforce the semicolon rule on tslint to fix that

Reply:
Pourquoi aller à l'encontre du linter? Pourquoi pas mettre des ; partout?